testing/ef-tests/src/models.rs

```diff
@@ -347,47 +347,52 @@ impl From<ForkSpec> for ChainSpec {
     }
 }
 
-impl From<ForkSpec> for reth_stateless::fork_spec::ForkSpec {
-    fn from(value: ForkSpec) -> Self {
-        match value {
-            ForkSpec::Frontier => reth_stateless::fork_spec::ForkSpec::Frontier,
-            ForkSpec::FrontierToHomesteadAt5 => {
-                reth_stateless::fork_spec::ForkSpec::FrontierToHomesteadAt5
-            }
-            ForkSpec::Homestead => reth_stateless::fork_spec::ForkSpec::Homestead,
-            ForkSpec::HomesteadToDaoAt5 => reth_stateless::fork_spec::ForkSpec::HomesteadToDaoAt5,
-            ForkSpec::HomesteadToEIP150At5 => {
-                reth_stateless::fork_spec::ForkSpec::HomesteadToEIP150At5
-            }
-            ForkSpec::EIP150 => reth_stateless::fork_spec::ForkSpec::EIP150,
-            ForkSpec::EIP158 => reth_stateless::fork_spec::ForkSpec::EIP158,
-            ForkSpec::EIP158ToByzantiumAt5 => {
-                reth_stateless::fork_spec::ForkSpec::EIP158ToByzantiumAt5
+impl From<ForkSpec> for reth_stateless::chain_spec::ChainSpec {
```

Comment:
@jsign have moved the From impl to ef-tests, so the host will be able to get a ef_tests::ForkSpec, then convert it to a reth_stateless::ChainSpec. Instead of ef_tests::ForkSpec to reth_stateless::ForkSpec to reth_stateless::ChainSpec.

This should be allowed because of Rust's orphan rule

Reply:
This should also avoid the issue of reth_stateless pulling in anything from ef-tests that we had before since its not no_std

Reply:
I think the guest program would never see ForkSpec, can you say why it would need to pull it in?

Reply:
Oh the ChainConfig is an enum that contains a `ForkSpec`

Reply:
My thinking is that we send over reth_stateless::ChainSpec to the guest program and not ChainConfig -- so ForkSpec is never needed. Does that work?

Reply:
> Oh the ChainConfig is an enum that contains a `ForkSpec` (in the benchmarking repo)

Yep, exactly. `ForkSpec` is the input type that the zkVM receives.

> My thinking is that we send over reth_stateless::ChainSpec to the guest program and not ChainConfig -- so ForkSpec is never needed. Does that work?

Uhm, I think should be fine -- assuming the serialization of that works.

So that would mean [this logic](https://github.com/eth-act/zkevm-benchmark-workload/blob/master/ere-guests/libs/src/chainconfig.rs#L18) would stop being done in the guest programs, and it would be done on the witness-generator, which would be fine with not asking no_std?

Reply:
Though let me think about the conversion working point

Reply:
Sounds good -- though this may require modifying quite a lot of things, perhaps my previous PR with ChainConfig may be a better route, not sure